core/iwasm/libraries/debug-engine/debug_engine.c

```diff
@@ -1194,24 +1211,30 @@ wasm_debug_instance_mmap(WASMDebugInstance *instance, uint32 size,
     if (!exec_env)
         return 0;
 
-    module_inst = (WASMModuleInstance *)exec_env->module_inst;
+    if (instance->exec_mem_info.start_offset == 0) {
+        return 0;
+    }
 
-    /* TODO: malloc in wasi libc maybe not be thread safe, we hope LLDB will
-             always ask for memory when threads stopped */
-    offset = wasm_runtime_module_malloc((wasm_module_inst_t)module_inst, size,
-                                        &native_addr);
-    if (!offset)
+    if ((uint64)(instance->exec_mem_info.current_pos
+                 - instance->exec_mem_info.start_offset + size)
+        <= (uint64)instance->exec_mem_info.size) {
+        offset = instance->exec_mem_info.current_pos;
+        instance->exec_mem_info.current_pos += size;
```

Comment:
Directly adding the position instead of allocating memory, there is potential issue in multi-thread, should we initialize the preserved memory as an heap, and allocate memory from the heap?

Reply:
The 'allocation' can only happen when the execution is stopped, so only the debug control thread of this debug instance can access this struct, that's why I add the position simply and without lock here.

Of course, we can use an allocator, but according to my observation, the lldb will never request to free this space, so I suggest to use the simplest solution, what's your opinion on this?

Reply:
Yes, that should be OK.